### PR TITLE
PI-18 Google Sign-in 

### PR DESCRIPTION
**Pull request by kris-krsh02**

### CODE REVIEW

File: lib/user_auth.dart [hunk 1]
```diff
@@ -18,4 +19,33 @@ class UserAuth {
...
+
+Future<User?> signInWithGoogle() async {
+final GoogleSignIn googleSignIn = GoogleSignIn(
+      clientId: '997688084433-qiahs2fdcekq042vg9k7dl7mebo58v7d.apps.googleusercontent.com',
+    );
+
+    // Attempt to sign in the user with Google
+    final GoogleSignInAccount? googleUser = await googleSignIn.signIn();
+    print('sign in with google');
+
```
Comment (campcoxconsultants) [inline lib/user_auth.dart]: handle if null

File: lib/user_auth.dart [hunk 2]
```diff
@@ -18,4 +19,33 @@ class UserAuth {
...
+    // Obtain the authentication tokens from the Google sign-in process
+    final GoogleSignInAuthentication? googleAuth = await googleUser!.authentication;
+   
+
```
Comment (campcoxconsultants) [inline lib/user_auth.dart:36]: if google auth == null return

File: lib/user_auth.dart [hunk 3]
```diff
@@ -18,4 +19,33 @@ class UserAuth {
...
+    // Create a new credential for signing in with Firebase
+    final OAuthCredential credential = GoogleAuthProvider.credential(
+      accessToken: googleAuth!.accessToken,
```
Comment (campcoxconsultants) [inline lib/user_auth.dart]: requires At least one of accessToken or idToken so add a guard if before

File: lib/user_auth.dart (hunk 2 above)
Comment (kris-krsh02) [reply lib/user_auth.dart:36]: Dart gives me a problem here, stating that googleAuth is non-nullable. Could it really be null if googleUser exists (is not null)?

**Comment by kris-krsh02** [comment]
image

File: lib/user_auth.dart (hunk 2 above)
Comment (campcoxconsultants) [reply lib/user_auth.dart:36]: hmm ok, then remove the ? on line 34?

**Review by campcoxconsultants — APPROVED** [review]
this looks good, if we can fix the one part, this is great

File: lib/user_auth.dart (hunk 2 above)
Comment (kris-krsh02) [reply lib/user_auth.dart:36]: Removed it, and also removed the if google auth == null return since it is not needed anymore

**Pull request merged**